src/net/rpc.rs

```diff
@@ -3397,6 +3502,186 @@ mod test {
         );
     }
 
+    /// Expecting the peer server's block height to be 1 while its peer's block height is 3.
+    /// This leads to a response where the peer server returns false for `matches_peers` as well as
+    /// the percent of blocks it has.
+    #[test]
+    #[ignore]
+    fn test_rpc_get_health_false() {
+        test_rpc(
+            "test_rpc_get_health_false",
+            40010,
+            40011,
+            50010,
+            50011,
+            |ref mut peer_client,
+             ref mut convo_client,
+             ref mut peer_server,
+             ref mut convo_server| {
+                // set up block stats for peer server
+                let peer_server_sortdb = peer_server.sortdb.take().unwrap();
+
+                if peer_server.network.inv_state.is_none() {
+                    peer_server.network.init_inv_sync(&peer_server_sortdb);
+                }
+                let peer_client_nk = peer_client.to_neighbor().addr;
+
+                if let Some(ref mut inv_state) = peer_server.network.inv_state {
+                    if inv_state.get_stats(&peer_client_nk).is_none() {
+                        inv_state.add_peer(peer_client_nk.clone(), true);
+                    }
+                    inv_state
+                        .get_stats_mut(&peer_client_nk)
+                        .unwrap()
+                        .inv
+                        .num_sortitions = 3;
+                }
+                peer_server.sortdb = Some(peer_server_sortdb);
+                convo_client.new_get_health()
+            },
+            |ref http_request, ref http_response, ref mut peer_client, ref mut peer_server| {
+                let req_md = http_request.metadata().clone();
+                match http_response {
+                    HttpResponseType::GetHealthError(response_md, data) => {
+                        let get_health_error_resp: GetHealthResponse =
+                            serde_json::from_value(data.clone()).unwrap();
+                        assert_eq!(get_health_error_resp.matches_peers, false);
+                        assert_eq!(get_health_error_resp.percent_of_blocks_synced, 33);
+                        true
+                    }
+                    _ => {
+                        error!("Invalid response: {:?}", &http_response);
+                        false
+                    }
+                }
+            },
+        );
+    }
+
+    /// The block height of the peer server and its peer should match, so `matches_peers` is true.
+    #[test]
+    #[ignore]
+    fn test_rpc_get_health_true() {
+        test_rpc(
+            "test_rpc_get_health",
```

Comment:
This string is going to need to be different for each unit test, since it ultimately gets used to create a directory.  Tests running in parallel will clobber each other.  Also, the numbers below are port numbers -- they need to be different from other unit tests' port numbers in order to avoid `EADDRINUSE` errors.